### DISCUSSION TITLE
[Suggestion] Sort *.ledger file by time of transaction

### DISCUSSION THREAD

**Discussion by TMaddox** [ananthakumaran/paisa#199] [Ideas]
How about a sort feature (especially by time) for the ledger file(s)? I am trying to import from several *.csv files (from several accounts) and merging them is quite a challenge. Thanks!

**Comment by ananthakumaran** [comment]
Have you tried using [multiple files / include](https://paisa.fyi/reference/journal/#journal) feature. You can then use a ledger file per month or year and it will be much easier to manage.

I am converting this to a discussion, issue is for bug only

**Comment by TMaddox** [comment]
My apologies about the Bug/Discussion thing, should have read your readme first.

I will definitely use that feature and it makes sense to have one per account, you are right. Sorting per timestep would nevertheless be a valuable addition IMHO.

To add to the discussion: How about a duplicate finder? I was working on a project before (planned on open sourcing once complete) before I discovered ledger and your project. Here is a small snipped of what I did on that topic:

```
def identify_self_transfers(transactions_df, day_threshold=7):
    # Filter income and expense transactions
    income_df = transactions_df[transactions_df["transaction_type"] == TransactionType.INCOME]
    expense_df = transactions_df[transactions_df["transaction_type"] == TransactionType.EXPENSE]

    # Identify potential pairs
    pairs = []
    for index, income_transaction in income_df.iterrows():
        potential_matches = expense_df[
            (expense_df["amount"] == income_transaction["amount"])
            & (
                abs(
                    (
                        expense_df["date_transaction"] - income_transaction["date_transaction"]
                    ).dt.days
                )
                <= day_threshold
            )
        ]
        for index_match, match in potential_matches.iterrows():
            pairs.append((index, index_match))

    return pairs
```
